otherlibs/systhreads/st_stubs.c

```diff
@@ -948,3 +965,33 @@ static st_retcode caml_threadstatus_wait (value wrapper)
 
   CAMLreturnT(st_retcode, retcode);
 }
+
+/* Set the current thread's name. */
+CAMLprim value caml_current_thread_set_name(value name)
+{
+#if defined(_WIN32)
+  wchar_t *thread_name = caml_stat_strdup_to_utf16(String_val(name));
+  SetThreadDescription(GetCurrentThread(), thread_name);
+  caml_stat_free(thread_name);
+
+  // We are using both methods.
+  // See: https://github.com/ocaml/ocaml/pull/13504#discussion_r1786358928
+  pthread_setname_np(pthread_self(), String_val(name));
+#elif defined(HAS_PRCTL)
+  prctl(PR_SET_NAME, String_val(name));
+#elif defined(HAS_PTHREAD_SETNAME_NP)
+#  if defined(__APPLE__)
+  pthread_setname_np(String_val(name));
+#  elif defined(__NetBSD__)
+  pthread_setname_np(pthread_self(), "%s", String_val(name));
+#  else
+  pthread_setname_np(pthread_self(), String_val(name));
+#  endif
+#elif defined(HAS_PTHREAD_SET_NAME_NP)
+  pthread_set_name_np(pthread_self(), String_val(name));
+#else
+  caml_invalid_argument("set thread name not implemented");
```

Comment:
Why not just make the case where it's not implemented a noop instead of raising an exception?

This comes up when compiling to exotic platforms like unikernels, and results in the program raising an exception at start-of-day. But in practise, if there's no interface to set a thread name, then there's no interface to query a thread name either, and you might as well continue the execution of the program. All that happens at this point is that the library runtime has to stub this call out.

Reply:
Sure thing. Raising an exception is a common pattern when implementations are missing on e.g. unix for windows but since this is for debug and won't impact runtime, it makes sense to silently go ahead. I have added a `stderr` log.

Reply:
Emitting `stderr` warnings for something that neither program authors nor program users have much control over is not a terribly popular UI pattern. Within the OCaml runtime, it is controlled by a `caml_runtime_warnings_active()` on the C side, and `Sys.runtime_warnings_enabled ()` on the OCaml side. You should check `caml_runtime_warnings_active()` and not print anything if it is false.

Reply:
Ok. Just pushed the changes